.github/workflows/rust.yml

```diff
@@ -23,4 +22,6 @@ jobs:
       - name: Install mpirun
         run: cargo install --force cargo-mpirun
       - name: Run tests
-        run: cargo test --verbose
\ No newline at end of file
+        env:
+          RUST_BACKTRACE: 1
+        run: cargo test --verbose -- --test-threads=1
```

Comment:
Some of the multithreading tests fail with github actions. I think this is because the github actions server only has a very limited number of threads, but normally the unit tests of each file are run in parallel. That causes problems if tests with more threads needed  as the system has run at the same.

Reply:
That makes sense. Does this solve the problem?

Reply:
Yes. No checks are all green :)